code/game/machinery/machinery.dm

```diff
@@ -587,6 +587,8 @@ Class Procs:
 	if(!anchored && machine_flags & FIXED2WORK)
 		return user << "<span class='warning'>\The [src] must be anchored first!</span>"
 
+	return exchange_parts(user, O)

```

Comment:
Check the istype on this instead of plain returning it, having the proc call every time is needless.
And then you can remove the istype from the proc eeeeeeh